source/routing/raptor_api.cpp

```diff
@@ -51,6 +51,126 @@ namespace navitia { namespace routing {
+            return (total_nb_journeys < min_nb_journeys.get()) && (nb_try <= max_nb_raptor_call);
+        }
+        if (total_nb_journeys < min_nb_journeys.get()) {
+            // Case 2: we continue searching journeys until min_nb_journeys is reached
+            return true;
+        }
+        // Case 3: we have already enough journeys, but the search datetime is still inside of the time frame
+        //         we keep searching until the end of time frame
+        return is_inside(request_date_secs, timeframe_duration.get());
+    }
+    if (! timeframe_duration) {
+        // Case 4: Neither time duration nor min_nb_journeys are given, we call only once Raptor
+        return false;
+    }
+    // Case 5: no min_nb_journeys is given, we find all journeys inside of the time frame
+    return is_inside(request_date_secs, timeframe_duration.get());
+}
+
+/**
+ * @brief Culling excessive journeys if we exceed limits
+ */
+static void culling_excessive_journeys(const boost::optional<uint32_t>& min_nb_journeys,
+                                       const boost::optional<uint32_t>& timeframe_duration,
+                                       const bool clockwise,
+                                       JourneySet& journeys) {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+
+    if (timeframe_duration && journeys.size() > 0) {
+
+        routing::JourneyCmp journey_cmp{clockwise};
+        // filter journeys by end datetime until min_nb_journeys
+        std::vector<routing::Journey> sorted{journeys.begin(), journeys.end()};
+        std::sort(sorted.begin(), sorted.end(), journey_cmp);
+
+        auto is_inside = [clockwise](const routing::Journey& j, DateTime dt){
+            return clockwise ? j.departure_dt < dt : j.arrival_dt > dt;
+        };
+
+        uint32_t count = 0;
+        boost::for_each(sorted, [&](const routing::Journey& j) {
+
+            // Erase journey if
+            // min_nb_journeys is deactivated and journey exceed the time frame limit
+            // or
+            // min_nb_journeys exist and Nb journeys is greater than min_nb_journeys criteria
+            // and journey exceeds the time frame limit
+            if ( (!min_nb_journeys && !is_inside(j, timeframe_duration.get())) ||
+                 (min_nb_journeys && (count >= min_nb_journeys.get()) && !is_inside(j, timeframe_duration.get())) )
+            {
+                journeys.erase(j);
+            }
+            else {
+                ++count;
+            }
+        });
+        LOG4CPLUS_DEBUG(logger, "after culling excessive journeys: " << journeys.size() << " solution(s) left");
+    }
+}
+
+/**
+ * @brief Update timeframe_duration in raptor referential
+ */
+static void update_timeframe_parameters(const uint32_t request_date_secs,
+                                        const bool clockwise,
+                                        boost::optional<uint32_t>& timeframe_duration)
```

Comment:
From what I understand (naming is hard here) this looks ugly: input is a duration, output is a datetime...
We are changing :carrot: in :potato: , and I don't like :stew: